src/core/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,298 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import { IconButton } from '../Buttons';
+import SvgChevronLeft from '@itwin/itwinui-icons-react/cjs/icons/ChevronLeft';
+import SvgChevronRight from '@itwin/itwinui-icons-react/cjs/icons/ChevronRight';
+import '@itwin/itwinui-css/css/carousel.css';
+
+export type CarouselProps = {
+  currentIndex?: number;
+  autoPlay?: boolean;
+} & React.ComponentPropsWithoutRef<'div'>;
+
+const CarouselContext = React.createContext<
+  | {
+      currentIndex: number;
+      setCurrentIndex: (index: number | ((old: number) => void)) => void;
+      slideCount: number;
+      setSlideCount: (length: number | ((old: number) => void)) => void;
+      keysPressed: Record<string, boolean>;
+    }
+  | undefined
+>(undefined);
+
+export const Carousel = (props: CarouselProps) => {
+  const { currentIndex: userCurrentIndex = 0, children, ...rest } = props;
+
+  useTheme();
+
+  const [currentIndex, setCurrentIndex] = React.useState(userCurrentIndex);
+  const [slideCount, setSlideCount] = React.useState(0);
+
+  const [keysPressed, setKeysPressed] = React.useState<Record<string, boolean>>(
+    {},
+  );
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.altKey || event.ctrlKey || event.metaKey) {
```

Comment:
As you started...what about `shift` then? :D

Reply:
And seems like we should have some util for all our events to ignore those additional keys :/ but out of this scope

Reply:
added shiftKey